samples/cellular/modem_shell/src/cloud/cloud_lwm2m.c

```diff
@@ -34,8 +34,8 @@ BUILD_ASSERT(sizeof(CONFIG_MOSH_LWM2M_PSK) > 1, "LwM2M pre-shared key (PSK) must
 #define MEMORY_TOTAL_RID 21
 
 #define IMEI_LEN 15
-static uint8_t imei_buf[IMEI_LEN + sizeof("\r\nOK\r\n")];
-#define ENDPOINT_NAME_LEN (IMEI_LEN + sizeof(CONFIG_MOSH_LWM2M_ENDPOINT_PREFIX) + 1)
+static uint8_t imei_buf[IMEI_LEN + 1];
+#define ENDPOINT_NAME_LEN (sizeof(CONFIG_MOSH_LWM2M_ENDPOINT_PREFIX) + IMEI_LEN)
```

Comment:
I just have to assume these are fine. Some comment somewhere (in commit etc.) would be good.

Reply:
There was a compilation warning from a `snprintk()` which uses these. I can't remember where the original code came from, but those seemed to be simply incorrect. The `_LEN` defines should match the length of the string without the terminating NUL and the structs should have `+ 1` to have room for the termination. And, the IMEI buffer does not need to have room for "OK" and CRLFs. Maybe the original code queried the IMEI directly from the modem using an AT command, I don't know.

I can add a note in the commit message.

Reply:
This explanation is fine. At least there is some note about the change somewhere :smile: